common/src/test/java/org/cloudfoundry/identity/uaa/oauth/ClientAdminEndpointsTests.java

```diff
@@ -694,4 +696,91 @@ public void testErrorHandler() throws Exception {
+        assertFalse(created.isAutoApprove("foo.write"));
+    }
+    
+    private static void assertSetEquals(Collection<?> a, Collection<?> b) {
+        assertTrue("expected "+a+" but was "+b, a == null && b == null || a != null && b != null && a.containsAll(b) && b.containsAll(a));
+    }
+    
+    @Test
+    public void testCreateClientWithAutoapproveScopesTrue() throws Exception {
+        
+        List<String> scopes = Arrays.asList("foo.read","foo.write");
+        List<String> autoApproveScopes = Arrays.asList("true");
+        input.setScope(scopes);
+        detail.setScope(scopes);
+        input.setAutoApproveScopes(autoApproveScopes);
+        detail.setAutoApproveScopes(autoApproveScopes);
+        detail.setAuthorizedGrantTypes(input.getAuthorizedGrantTypes());
+        ClientDetails result = endpoints.createClientDetails(input);
+        assertNull(result.getClientSecret());
+        ArgumentCaptor<BaseClientDetails> clientCaptor = ArgumentCaptor.forClass(BaseClientDetails.class);
+        Mockito.verify(clientRegistrationService).addClientDetails(clientCaptor.capture());
+        BaseClientDetails created = clientCaptor.getValue();
+        assertSetEquals(autoApproveScopes, created.getAutoApproveScopes());
+        assertTrue(created.isAutoApprove("foo.read"));
+        assertTrue(created.isAutoApprove("foo.write"));
+    }
+    
+    
+    @Test
+    public void testUpdateClientWithAutoapproveScopesList() throws Exception {
+        List<String> scopes = Arrays.asList("foo.read","foo.write");
+        List<String> autoApproveScopes = Arrays.asList("foo.read");
+        
+        input.setScope(scopes);
+        detail.setScope(scopes);
+        detail.setAutoApproveScopes(autoApproveScopes);
+        
+        Mockito.when(clientDetailsService.retrieve(input.getClientId())).thenReturn(
+                new BaseClientDetails(input));
+        ClientDetails result = endpoints.updateClientDetails(detail, input.getClientId());
+        assertNull(result.getClientSecret());
+        ArgumentCaptor<BaseClientDetails> clientCaptor = ArgumentCaptor.forClass(BaseClientDetails.class);
+        Mockito.verify(clientRegistrationService).updateClientDetails(clientCaptor.capture());
+        BaseClientDetails updated = clientCaptor.getValue();
+        assertSetEquals(autoApproveScopes, updated.getAutoApproveScopes());
+        assertTrue(updated.isAutoApprove("foo.read"));
+        assertFalse(updated.isAutoApprove("foo.write"));
+    }
+    
+    @Test
+    public void testUpdateClientWithAutoapproveScopesTrue() throws Exception {
+        List<String> scopes = Arrays.asList("foo.read","foo.write");
+        List<String> autoApproveScopes = Arrays.asList("true");
+        
+        input.setScope(scopes);
+        detail.setScope(scopes);
+        detail.setAutoApproveScopes(autoApproveScopes);
+        
+        Mockito.when(clientDetailsService.retrieve(input.getClientId())).thenReturn(
+                new BaseClientDetails(input));
```

Comment:
This is faulty, if you look at the constructor, it doesn't copy the autoapprove field for you

	public BaseClientDetails(ClientDetails prototype) {
		this();
		setAccessTokenValiditySeconds(prototype.getAccessTokenValiditySeconds());
		setRefreshTokenValiditySeconds(prototype
				.getRefreshTokenValiditySeconds());
		setAuthorities(prototype.getAuthorities());
		setAuthorizedGrantTypes(prototype.getAuthorizedGrantTypes());
		setClientId(prototype.getClientId());
		setClientSecret(prototype.getClientSecret());
		setRegisteredRedirectUri(prototype.getRegisteredRedirectUri());
		setScope(prototype.getScope());
		setResourceIds(prototype.getResourceIds());
	}